sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -56,14 +56,11 @@ private[hive] class SparkExecuteStatementOperation(
   private var statementId: String = _
 
   private lazy val resultSchema: TableSchema = {
-    if (result == null || result.queryExecution.analyzed.output.size == 0) {
+    if (result == null || result.queryExecution.analyzed.output.isEmpty) {
```

Comment:
Done. That's much better. I fixed those to use `result.schema` as you mentioned. Especially, `logInfo(..)`, it changed to print the real result schema instead of attribute.
- Before: `Result Schema: List(1#5, a#6)`
- After: `Result Schema: StructType(StructField(1,IntegerType,false), StructField(a,StringType,false))`